pkg/finder/fsfinder.go

```diff
@@ -55,6 +59,62 @@ func WithDepth(depthVal int) FSFinderOptions {
 	}
 }
 
+// AddFiles adds unchecked files with specified formats to FSFinder (adds support for extensionless files)
+func AddFiles(files []string) FSFinderOptions {
+	return func(fsf *FileSystemFinder) {
+		if len(files) == 0 {
+			return
+		}
+		fsf.UncheckedFiles= make(map[string]string)
+		for i := range files {
+			kv := strings.Split(files[i], ":")
+			if len(kv) != 2 {
+				continue
+			}
+			// Creating <file name/dir>:<file type> mapping to link extensionless files with their actual format
+			fsf.UncheckedFiles[filepath.Clean(kv[0])] = kv[1]
+		}
+	}
+}
+
+// StoreConfig stores config file path to read in data later
+func StoreConfig(file string) FSFinderOptions {
+	return func(fsf *FileSystemFinder) {
+		fsf.ConfigFile = file
+	}
+}
+
+// ReadConfig reads in a config file in yaml format.
+// It converts the stored files and typings to mapped KVpairs,
+// similar to AddFiles in function and purpose.
+func ReadConfig(file string, fsf FileSystemFinder) (map[string]string, error) {
+	if fsf.UncheckedFiles == nil {
+		fsf.UncheckedFiles= make(map[string]string)
+	}
+	configFile, err := os.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read in config file using path %s", file)
+	}
+
+	// Create a map to store the unmarshaled data
+	data := make(map[string][]interface{})
+
+	// Unmarshal the YAML data into the map
+	err = yaml.Unmarshal(configFile, &data)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to read in config file data from %s", file)
+	}
+
+	// Storing the imported data as 'additional files' for findOne to access
+	for key, value := range data {
+		for _, val := range value {
+			strVal := val.(string)
+			fsf.UncheckedFiles[filepath.Clean(strVal)] = key
```

Comment:
I would define an alias type and do this

```go
type mapFiles map[string][string]

func (m *mapFiles) Add(path, value string) {
      m[filepath.Clean(path) = value
}

func (m *mapFiles) Has(key string) bool {
      _, ok := m[filepath.Clean(key)
      return ok
}
```

And use here and everywhere

```suggestion
			fsf.UncheckedFiles.Add(strVal, key)
```


```diff
-                                 if _, ok := fsf.ExcludeFileTypes[walkFileExtension]; ok {
+                                 if  fsf.ExcludeFileTypes.Has(walkFileExtension) {